quipucords/api/serializers.py

```diff
@@ -39,3 +39,46 @@ def validate(self, attrs):
             raise ValidationError(_('ssh_keyfile, %s, is not a valid file'
                                     ' on the system.' % (ssh_keyfile)))
         return attrs
+
+
+class NetworkProfileSerializer(ModelSerializer):
+    """Serializer for the NetworkProfile model"""
+    class Meta:
+        model = NetworkProfile
+        fields = '__all__'
+
+    @staticmethod
+    def validate_name(name):
```

Comment:
Seems like we should have this for Host Credentials as well.

Reply:
This is based on what HostCredentials does. The advantage to the field-by-field approach is that if you're doing a partial update, Django won't check the fields that aren't part of the update, and therefore won't automatically fail the update. I will update HostCredentials too.

Reply:
Yes, I was suggesting we add the validate_name such that we check that its printable to Host Credentials serializer.